src/game_inventory.cpp

```diff
@@ -788,10 +788,11 @@ class read_inventory_preset: public pickup_inventory_preset
                 }
                 const auto &book = get_book( loc );
                 if( book.skill && p.get_skill_level_object( book.skill ).can_train() ) {
-                    return string_format( _( "%s to %d" ), book.skill->name(), book.level );
+                    return string_format( _( "%s to %d (%d)" ), book.skill->name(), book.level,
+                                          g->u.get_skill_level( book.skill.obj().ident() ) );
```

Comment:
`p.get_skill_level_object()` above yields what you need, and it doesn't reference `g->u`. You could stash and reuse the result.

Reply:
you're right about not needing `g->u`, but I don't see `p.get_skill_level_object()` gets me what I'm after?

I did change to `p.get_skill_level( book.skill )`

Reply:
`SkillLevel::level()` is precisely what you need. So something like:
```c++
const SkillLevel &skill = p.get_skill_level_object( book.skill );

... skill.level() ...
```
would do.

Reply:
I'm afraid I can't understand what you're trying to tell me. :(

Reply:
Then I'll try to express myself in code (:

```c++
if( book.skill ) {
    const SkillLevel &skill = p.get_skill_level_object( book.skill );
    
    if( skill.can_train() ) {
	return string_format( _( "%s to %d (%d)" ), book.skill->name(), book.level, skill.level() );
    }
}

return std::string();
```

Reply:
I really appreciate your patience with me on this. C++ code is extra alien for me, so sometimes my brain just balks at trying to understand some things.

Reply:
No worries, I'm happy to help.